ui/packages/catalog-ui-search/src/main/webapp/component/location-new/utils/dms-utils.js

```diff
@@ -264,8 +268,82 @@ function validateDms(dms) {
     return { valid: valid, error: error };
 }
 
+/*
+ *  Decimal degrees -> DMS conversion utils
+ */
+function ddToDmsCoordinateLat(dd, secondsPrecision = DEFAULT_SECONDS_PRECISION) {
+    if (isNaN(dd)) {
+        return undefined;
+    }
+
+    const direction = dd >= 0 ? 'N' : 'S';
+    return ddToDmsCoordinate(dd, direction, LAT_DEGREES_DIGITS, secondsPrecision);
+}
+
+function ddToDmsCoordinateLon(dd, secondsPrecision = DEFAULT_SECONDS_PRECISION) {
+    if (isNaN(dd)) {
+        return undefined;
+    }
+
+    const direction = dd >= 0 ? 'E' : 'W';
+    return ddToDmsCoordinate(dd, direction, LON_DEGREES_DIGITS, secondsPrecision);
+}
+
+function ddToDmsCoordinate(dd, direction, degreesPad, secondsPrecision = DEFAULT_SECONDS_PRECISION) {
+    const ddAbsoluteValue = Math.abs(dd);
+    const degrees = Math.trunc(ddAbsoluteValue);
+    const degreeFraction = ddAbsoluteValue - degrees;
+    const minutes = Math.trunc(60 * degreeFraction);
+    const seconds = 3600 * degreeFraction - 60 * minutes;
+    const secondsRounded = roundTo(seconds, secondsPrecision);
+    return {
+        coordinate: `${pad(degrees, degreesPad)}°${pad(minutes, 2)}'${padDecimal(secondsRounded, 2)}"`,
+        direction: direction
+    };
+}
+
+function roundTo(num, sigDigits) {
+    const scaler = 10 ** sigDigits;
+    return Math.round(num * scaler) / scaler;
+}
+
+function pad(num, width) {
```

Comment:
Actually can we use [padStart()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart)?

Reply:
if you do go with bellcc's `numStr = padValue.repeat(width - numStr.length);` is missing `+ numStr;`

Reply:
@jrnorth Yeah `padStart()` would be great because we don't even need to do any math. It's just  `return num.toString().padStart(width, '0')`. You wouldn't even need to do any checks

Good find!

<img width="194" alt="screen shot 2018-08-22 at 7 38 24 pm" src="https://user-images.githubusercontent.com/7646577/44499834-01eb0800-a643-11e8-9b50-ceb1b1d5b9b9.png">

Reply:
Just wondering if we would need to do anything compatibility-wise since it is part of ECMAScript 2017. @djblue @andrewkfiedler

Reply:
We use `@babel/polyfill` to bring in those methods, we should be 👍